### DISCUSSION TITLE
Accept Codeium snippet suggestion with Tab?

### DISCUSSION THREAD

**Discussion by IvanXh0** [LazyVim/LazyVim#1796] [Q&A] [answered]
Hi guys, so basically I want to map my Tab key whenever there is a code suggestion from Codeium available and I'm in insert mode, to accept it with Tab. This is my current config: 

```
return {
  {
    "Exafunction/codeium.vim",
    lazy = false,
    config = function()
      vim.keymap.set("i", "<Tab>", function()
        return vim.fn["codeium#Accept"]()
      end, { expr = true })
      vim.keymap.set("i", "<c-;>", function()
        return vim.fn["codeium#CycleCompletions"](1)
      end, { expr = true })
      vim.keymap.set("i", "<c-,>", function()
        return vim.fn["codeium#CycleCompletions"](-1)
      end, { expr = true })
      vim.keymap.set("i", "<c-x>", function()
        return vim.fn["codeium#Clear"]()
      end, { expr = true })
    end,
  },
}

```

The current behaviour is, even though there is a code suggestion, when pressing Tab it just indents. However, if I remap it to  `<C-g>` lets say, it works fine. Any way of fixing this? Thank you guys.

**Comment by dpetka2001** [comment, marked as answer]
```lua
return {
  {
    "L3MON4D3/LuaSnip",
    keys = {
      { "<tab>", mode = { "i" }, false },
    },
  },

  {
    "Exafunction/codeium.vim",
    lazy = false,
    config = function()
      vim.keymap.set("i", "<tab>", function()
        return vim.fn["codeium#Accept"]()
      end, { expr = true })
      vim.keymap.set("i", "<c-;>", function()
        return vim.fn["codeium#CycleCompletions"](1)
      end, { expr = true })
      vim.keymap.set("i", "<c-,>", function()
        return vim.fn["codeium#CycleCompletions"](-1)
      end, { expr = true })
      vim.keymap.set("i", "<c-x>", function()
        return vim.fn["codeium#Clear"]()
      end, { expr = true })
    end,
  },
}
```
this worked for me, but you're losing the `<tab>` key for Luasnip jumping to nodes. You should probably remap this to a different key if you'd like to use the Luasnip functionality.

**Reply by IvanXh0** [reply]
Man oh man, I assumed that LuaSnip was the issue and tried remapping the keybinds, changing it to not work in insert mode, nothing really worked, but this fixes it for me. You're amazing, thank you! Marking this as solved.

**Reply by stephanbrez** [reply]
This no longer works in the latest lazyvim 14x with blink.cmp. Any ideas?

**Reply by folke** [reply]
`vim.g.ai_cmp = false`

**Reply by stephanbrez** [reply]
Thanks for the super fast response! It worked thank you!!

Wondering... is there a way to get the codeium suggestions inside of blink with tab, the same way it works with the supermaven extra?

**Reply by folke** [reply]
There is yes: `vim.g.ai_cmp = true`

**Reply by dpetka2001** [reply]
This post is talking about `codeium.vim` not `codeium.nvim` which is included in the Extras. I don't believe `codeium.vim` has sources for the completion engines.

**Reply by folke** [reply]
oh right, missed that.

**Reply by stephanbrez** [reply]
Oh cool, you can use codeium.**nvim** from Extras and make it behave like codeium.**vim** with `vim.g.ai_cmp = false`, or you can leave it as true and set  `['<Tab>'] = { 'select_and_accept' },` in blink opts to get a completion menu and tab completion.

Kinda makes using the .vim version unecessary now :smile: . Thanks for the help!
